src/core/MeshReflectorMaterial.tsx

```diff
@@ -194,12 +192,12 @@ export const MeshReflectorMaterial: ForwardRefComponent<MeshReflectorMaterialPro
 
       useFrame(() => {
         const parent = (materialRef.current as any).parent || (materialRef.current as any)?.__r3f.parent?.object
-        if (!parent) return
+        if (!parent || !materialRef.current.visible || !parent.visible || !shouldUpdate.current) return
```

Comment:
I used `shouldUpdate` to avoid unnecessary instructions in `useFrame` which  is called unconditionally.

Three's `onBeforeRender` instead is called only when the renderer needs it (EG: it shouldn't be called if the mesh isn't part of the scene being rendered).